[draco] Can't link drakodec.lib with target_link_libraries, usage file missing

Hello, after installing this library in vcpkg, it doesn't show how to use it.

I am trying to use the [draco] decoding support of the [tinygltf] library with this simple program example (main.cpp):

> ```
> #include <nlohmann\json.hpp>
> 
> #define TINYGLTF_NO_STB_IMAGE
> #define TINYGLTF_NO_STB_IMAGE_WRITE
> #define TINYGLTF_IMPLEMENTATION
> #define TINYGLTF_NO_INCLUDE_JSON
> #define TINYGLTF_ENABLE_DRACO
> #include <tiny_gltf.h>
> 
> int main()
> {
>   using namespace tinygltf;
> 
>   Model model;
>   TinyGLTF loader;
>   std::string err;
>   std::string warn;
> 
>   if (!loader.LoadBinaryFromFile(&model, &err, &warn, "models/viking_room.glb"))
>   {
>     throw std::runtime_error(warn + "\n" + err);
>   }
> 
>   return EXIT_SUCCESS;
> }
> ```

CMakeLists.txt

> ```
> cmake_minimum_required (VERSION 3.8)
> 
> project(TinyGltfDraco)
> 
> add_executable(TinyGltfDraco main.cpp)
> 
> set_target_properties(TinyGltfDraco
> PROPERTIES
>   CXX_STANDARD 17
>   CXX_EXTENSIONS OFF
> )
> 
> find_package(nlohmann_json CONFIG REQUIRED)
> target_link_libraries(TinyGltfDraco PRIVATE nlohmann_json nlohmann_json::nlohmann_json)
> 
> find_path(TINYGLTF_INCLUDE_DIRS "tiny_gltf.h")
> target_include_directories(TinyGltfDraco PRIVATE ${TINYGLTF_INCLUDE_DIRS})
> ```

The project is built by Cmake using the vcpkg toolkit. Building with the Visual Studio 2019 compilers generates some linkage errors:

> [build]   main.cpp
> [build] main.obj : error LNK2019: unresolved external symbol "public: __cdecl draco::Options::Options(void)" (??0Options@draco@@QEAA@XZ) referenced in function "public: __cdecl draco::DracoOptions<enum draco::GeometryAttribute::Type>::DracoOptions<enum draco::GeometryAttribute::Type>(void)" (??0?$DracoOptions@W4Type@GeometryAttribute@draco@@@draco@@QEAA@XZ) [%path%\build\TinyGltfDraco.vcxproj]

It looks like Draco header files can be found automatically with no additions to CMakeLists, but dracodec.lib or draco.lib is not passed to the linker.

If I add these lines to CMakeLists:

> ```
> find_package(Draco CONFIG REQUIRED)
> target_link_libraries(TinyGltfDraco PRIVATE draco)
> ```

I got that error:

> [build] LINK : fatal error LNK1104: cannot open file 'draco.lib' [%path%\build\TinyGltfDraco.vcxproj]

But something like this works (for a debug build):

> ```
> find_package(Draco CONFIG REQUIRED)
> target_link_libraries(TinyGltfDraco PRIVATE "${draco_LIBRARY_DIR}/../debug/lib/dracodec.lib")
> ```

I am wondering if I am doing something wrong or the port is missing some exports?

**Additional context:**
Draco's github page says we need to link the draco_dec (https://github.com/google/draco#c-decoder-api) but it's probably a typo (https://stackoverflow.com/questions/62313136/how-to-include-the-draco-dec-library). I have both draco.lib and dracodec.lib working so I'm not sure what should be linked by default. Should these be separate targets, like: `target_link_libraries(TinyGltfDraco PRIVATE draco::draco)` or `target_link_libraries(TinyGltfDraco PRIVATE draco::dracodec)`?

--- a/ports/draco/disable-symlinks.patch
+++ b/ports/draco/disable-symlinks.patch
@@ -1,19 +1,18 @@
 diff --git a/cmake/draco_targets.cmake b/cmake/draco_targets.cmake
-index f2acc72..fb8ce4a 100644
+index f96dbb1..179b9f9 100644
 --- a/cmake/draco_targets.cmake
 +++ b/cmake/draco_targets.cmake
-@@ -102,10 +102,6 @@ macro(draco_add_executable)
- 
+@@ -103,9 +103,6 @@ macro(draco_add_executable)
    add_executable(${exe_NAME} ${exe_SOURCES})
+   target_compile_features(${exe_NAME} PRIVATE cxx_std_11)
  
 -  if(NOT EMSCRIPTEN)
 -    set_target_properties(${exe_NAME} PROPERTIES VERSION ${DRACO_VERSION})
 -  endif()
--
+ 
    if(exe_OUTPUT_NAME)
      set_target_properties(${exe_NAME} PROPERTIES OUTPUT_NAME ${exe_OUTPUT_NAME})
-   endif()
-@@ -337,15 +333,6 @@ macro(draco_add_library)
+@@ -340,14 +337,6 @@ macro(draco_add_library)
      set_target_properties(${lib_NAME} PROPERTIES PREFIX "")
    endif()
  
@@ -25,7 +24,6 @@ index f2acc72..fb8ce4a 100644
 -                                       ${DRACO_SOVERSION_MAJOR})
 -    endif()
 -  endif()
--
+ 
    if(BUILD_SHARED_LIBS AND (MSVC OR WIN32))
      if(lib_TYPE STREQUAL SHARED)
-       target_compile_definitions(${lib_NAME} PRIVATE "DRACO_BUILDING_DLL=1")

--- a/ports/draco/portfile.cmake
+++ b/ports/draco/portfile.cmake
@@ -3,8 +3,8 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO google/draco
-    REF bd1e8de7dd0596c2cbe5929cbe1f5d2257cd33db #v1.5.2
-    SHA512 6ae7e72a9f6f55563f8f612084d38bff1d2e10934fa84aad59538d323e59d205764ed364c753a55d80e9ffc7c17f542f6475b3f922edcb9085cbd83a942759d0
+    REF 1.5.3
+    SHA512 8575ea78e0d8025facddbd42453b0251387f4e31eb0854135e050fc26aaf0d28ed30ccc3f93578fdc6cdb50369c2ef735291f1f5fb60238b289e0ee019446e1d
     HEAD_REF master
     PATCHES
         fix-compile-error-uwp.patch
@@ -18,7 +18,7 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 
-vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake)
+vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake/${PORT})
 vcpkg_fixup_pkgconfig()
 
 # Install tools and plugins
@@ -29,10 +29,8 @@ vcpkg_copy_tools(
     AUTO_CLEAN
 )
 
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")
 
 vcpkg_copy_pdbs()
 
-# Handle copyright
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/draco/vcpkg.json
+++ b/ports/draco/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "draco",
-  "version": "1.5.2",
+  "version": "1.5.3",
   "description": " A library for compressing and decompressing 3D geometric meshes and point clouds. It is intended to improve the storage and transmission of 3D graphics.",
   "homepage": "https://github.com/google/draco",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2009,7 +2009,7 @@
       "port-version": 0
     },
     "draco": {
-      "baseline": "1.5.2",
+      "baseline": "1.5.3",
       "port-version": 0
     },
     "drlibs": {

--- a/versions/d-/draco.json
+++ b/versions/d-/draco.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b4f39e03fdf47c2c1e9a945b362ed52ec098a354",
+      "version": "1.5.3",
+      "port-version": 0
+    },
+    {
       "git-tree": "39effb9dbafcf8540c1e99bacc9a2701ab9f94a1",
       "version": "1.5.2",
       "port-version": 0

